File: app/src/main/java/com/course/mindorks/todoapp/adapter/NotesAdapter.java
package com.course.mindorks.todoapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.course.mindorks.todoapp.R;
import com.course.mindorks.todoapp.clicklisteners.ItemClickListeners;
import com.course.mindorks.todoapp.model.Notes;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    List<Notes> adapterList = new ArrayList<>();
    private ItemClickListeners itemClickListeners;

    public NotesAdapter(List<Notes> list, ItemClickListeners itemClickListeners) {
        this.adapterList = list;
        this.itemClickListeners = itemClickListeners;
    }

    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {

        // bind the data from list view to holder
        final Notes notes = adapterList.get(position);
        holder.textViewTitle.setText(notes.getTitle());
        holder.textViewDescription.setText(notes.getDescription());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                itemClickListeners.onClick(notes);
            }
        });
    }

    @Override
    public int getItemCount() {

        return adapterList.size();
    }


    static class ViewHolder extends RecyclerView.ViewHolder {

        TextView textViewTitle, textViewDescription;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            textViewTitle = itemView.findViewById(R.id.textViewTitle);
            textViewDescription = itemView.findViewById(R.id.textViewDescription);
        }
    }
}

File: app/src/main/java/com/course/mindorks/todoapp/adapter/NotesAdapter.kt
package com.course.mindorks.todoapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.course.mindorks.todoapp.R
import com.course.mindorks.todoapp.clicklisteners.ItemClickListeners
import com.course.mindorks.todoapp.model.Notes

class NotesAdapter(val list: List<Notes>,
                   val itemClickListeners: ItemClickListeners)
    : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesAdapter.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) {
        val notes = list[position]
        val title = notes.title
        val discreption = notes.description
        holder.textViewTitle.text = title
        holder.textViewDescription.text = discreption
        holder.itemView.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                itemClickListeners.onClick(notes)
            }

        })


    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewTitle: TextView = itemView.findViewById(R.id.textViewTitle)
        val textViewDescription: TextView = itemView.findViewById(R.id.textViewDescription)
    }
}